### DISCUSSION TITLE
(Outdated) Analyzing the current Ubuntu desktop theme situation: Problems and solutions

### DISCUSSION THREAD

**Discussion by Feichtmeier** [ubuntu/yaru#2677] [General]
## Introduction

To make it easier for designers and non-gnomers or non-gnome-techies to jump into the color problem discussion I will summarize the current situation of the Ubuntu desktop design/theme, it's design and technical problems or limitations and give a list of potential solutions.

### What UI components does the Ubuntu desktop use?

Roughly speaking it consists of two technologies for the UI:

- GTK, written in C which is used for creating apps
- the gnome-shell written in javascript which is used as a visual shell around apps and as a visual bridge between the user and the system

**Problem 1**: _Both of those technologies are disconnected from each other and created, designed and maintained by the GNOME project and not by Ubuntu alone._

### How does the GNOME project theme it's desktop?

Each of the two independent technologies feature their own theme. They are technically completely disconnected and both themes come bundled with their technology.
This means that there is a built-in theme for apps and another built-in theme for the desktop.

**Problem 2**: *There is no theming API neither included in GTK nor in gnome-shell, unlike for example in android!*

### How does Ubuntu theme it's desktop?

[Ubuntu](https://design.ubuntu.com/brand/colour-palette/) as a brand features a different color scheme (`#E95420`, `#FFFFFF`, `#000000`, `#772953`, `#77216F`, `#5E2750`, `#2C001E`, `#111111`,  `#333333`, `#AEA79F`, `#0e8620`) than the [GNOME project](https://developer.gnome.org/hig/stable/icon-design.html.en#palette) (`#3584e4`, `#33d17a`, `#f6d32d`, `#ff7800`, `#e01b24`, `#9141ac`, `#986a44`, `#c0bfbc`).

**Problem 3**: _Or to be concrete and less detailed in the gnome project everything that should be highlighted is featured in a vivid blue, whereas ubuntu mainly features orange and purple._

With the [return to full gnome](https://ubuntu.com/blog/growing-ubuntu-for-cloud-and-iot-rather-than-phone-and-convergence) the desire to have the Ubuntu desktop branded in Ubuntu colors and realizing Problem 1, 2 and 3 this project was created.
So in a very smart way to built two individual theme stacks, namely themes for the apps and themes for the system, plus a full icon and cursor theme (which won't be discussed here), this repository creates themes for the Ubuntu desktop for each release cycle.

Those themes can be used via settings in gnome, which are normally hidden from the user in the gnome desktop to overwrite the default gnome look (blue->orange+purple+green).

### What is a dark mode/theme, why are they used?

Possibly connected to how you print and draw on white paper in reality, UIs used to exist mainly with dark fonts, glyphs and symbols on a white or light background.
Recently, or actually since quiet some time, UIs or apps or systems can have their whole visual representation swapped from dark text on light background to light text on dark background.
This is useful and popular because how good you can see the elements on your screen are bound to the current lightning conditions in your room and the quality of your display.

So when you use your computer/smartphone in the evening or in the night, where your environment most likely sends less light onto your display the contrast of a white screen against the dark room becomes too high. This is very bad for the eyes and can result in an actual loss or at least reduction of eyesight quality.

![grafik](https://user-images.githubusercontent.com/15329494/107531710-64db0b80-6bbd-11eb-88c2-a2e12a5f20e3.png)

So this is probably better for your eyes when you work while your room is dark:
![grafik](https://user-images.githubusercontent.com/15329494/107532236-f5195080-6bbd-11eb-883d-87127f23e7e5.png)

### Flip the situation: light themes

The opposite problematic situation is to be negated when your room is super bright or the sun is shining into your eyes / onto the screen and your UI is dark:

![grafik](https://user-images.githubusercontent.com/15329494/107532958-b8018e00-6bbe-11eb-9b9a-6f32d837a98a.png)

This is when light themes kick in and are actually really helpful to still be able to see everything on your screen even when it is bright outside of the screen:

![grafik](https://user-images.githubusercontent.com/15329494/107533265-0f076300-6bbf-11eb-8f6a-e22b26aed2f9.png)

#### Android

In Android every theme you create with the theming API has a brightness property, i.e. "is this a light theme?" or "is this a dark theme?" which you can set.

Android apps can follow this user set system preference and can opt-in to use "system light/dark preference" and if they have a theme with the "light" and a theme with the "dark" attribute. 

When the user toggles the system theme, which includes android settings, the onscreen display, notifications and widgets, the apps that have opted in switch along side with the system theme.

if the user still wants a specific app to not follow this global theme, he can again opt-out an app with a selected "always dark" or "always light" option.

TLDR: android has it all :+1: 

![android](https://user-images.githubusercontent.com/15329494/107545027-1a608b80-6bcb-11eb-8237-92660bdfb183.gif)

#### iOS

Apple smartphone feature a similar option, but it is not as easily toggled, so you need to enter the system settings app. But the opt-in works like on android.

![ios](https://user-images.githubusercontent.com/15329494/107545874-0ff2c180-6bcc-11eb-8539-b7c58e4079ea.gif)

#### GNOME

The gnome desktop does not have a dark mode. The default design language "adwaita" is light with blue accents. Certain apps that could be classified as "media-apps" feature an inverted adwaita theme which can be opted-in with the "prefer dark" option.
The system theme (gnome-shell theme) is always dark:

![gnome](https://user-images.githubusercontent.com/15329494/107547840-3285da00-6bce-11eb-8c2f-42d262f03a40.png)

#### Ubuntu

In Ubuntu the dark mode situation is slightly better and slightly worse at the same time.
While there is a theme selector page put into gnome control center, it can not target the system theme. So even when users chose between a dark, a light and a mixed application theme, the system stays light all the time.

Resulting in
**Problem 4a**: _the user prefence for a dark or a light mode is not respected globally_
**Problem 4b**: _and the default application theme plus the default system theme are not really light themes and not really dark themes so they are neither optimized for the day nor optimized for the night!_

![ubuntu](https://user-images.githubusercontent.com/15329494/107548724-4e3db000-6bcf-11eb-8e76-d9137198f612.png)

## Technical limitations with theming the GNOME desktop in Ubuntu

Problem 2 described that no theming API exists in gnome DE so there is not system "accent color" that could be opted-in from some setting/preferences or similar things like in Windows10.
So the only possibility to bring Ubuntu's colors to the gnome desktop is to provide a gtk theme and a gnome-shell theme, which are not connected to each other.
GTK themes are basically a css file in a "themes" directory. Each widget in the gnome application toolkit can be styled with a selector.

Example
```css
button {
 background-color: $bg_color;
 color: $text_color;
}
```
When an application wants to use a widget outside of the toolkit there is no existent selector with which those widgets are themed. This means that the app developer needs to provide a custom stylesheet for this widget.
The default gnome application theme "adwaita" puts special variables on the top of the stylesheet which applications can look up and take them to style their applications. Roughly said those variables are
- the global background color (gray)
- the global text color (black)
- the primary accent color (blue)
- the success color (green)
- the destructive color (red)
So when an app has a newly created custom widget that is not part of the gnome toolkit they can look up the "global colors" in the stylesheet. (But this is not a really an theming API, it could be described as an "agreement" of colors.)

So in gnome the developer can decide when those custom stylesheets are loaded, either _before_ the system theme or _after_ the system theme. Resulting in ...

**Problem 5:** _no matter how carefully we style the gnome applications gnome app developers can always block this from happening, resulting in some parts of the app being themed by the gnome theme and other parts with the ubuntu colors!_

Quiet problematic example is the gnome control center (System settings app) where you always see the blu progress bubbles for the volume output:

![blue](https://user-images.githubusercontent.com/15329494/107560115-408f2700-6bdd-11eb-90e4-651a8123b898.gif)

This is especially problematic for ubuntu desktops default theme, which features a light window, crowned by the dark headerbar.
So when a theme puts a custom widget into the headerbar and looks up for the current global background color it won't find the background color of the headerbar!

So we end up with the following problem quiet often:

**Problem 6**: _the headerbar colors are not exported at the top of the stylesheet of gtk!_

Examples: https://github.com/ubuntu/yaru/issues/2358, https://github.com/ubuntu/yaru/issues/2526
![grafik](https://user-images.githubusercontent.com/15329494/93706763-04720180-fb2a-11ea-9c3e-7720d484f2ee.png)
![grafik](https://user-images.githubusercontent.com/15329494/93706777-181d6800-fb2a-11ea-9ecf-d17492384212.png)
![grafik](https://user-images.githubusercontent.com/15329494/93706779-210e3980-fb2a-11ea-8dc5-dc6335d30923.png)

#### Problem severity

Low

#### Possible solutions

- contribute more to the gnome project, they are actually quiet open minded
- use a different desktop
- create our own desktop

#### Problem severity

Medium

#### Possible solutions

- develop a theming api for the gnome desktop and contribute it upstream
- use a different desktop/create our own desktop
- use only the gnome shell and use a different toolkit for the applications which feature a true theming api

#### Problem severity

Very low, it is very common that different companies and or brands have a different color palette. It is actually one of the first things you think about when you create a new brand

####  Problem severity

High - android and ios define what are minimum features for a modern desktop and people got very used to dark/light modes and expect it form a modern operating system!

#### Possible solutions

- introduce a global dark theme prefence in gnome
- use the dark gnome shell theme (the system theme) as default
![grafik](https://user-images.githubusercontent.com/15329494/107562258-d9bf3d00-6bdf-11eb-97d9-26343a5dfa8e.png)

#### Problem severity

Medium - users not familiar to the history of Ubuntu themes are not known to the concept of "mixed" themes and would expect the theme to be either dark or light

#### Possible solutions

- introduce this third "mixed" style on android and ios
- remove the mixed themed and provide only dark and light application themes, but having the system theme always dark, since it is less often displayed on your screen and a dark popup above a light popup is less annoying than the opposite

![onlylight](https://user-images.githubusercontent.com/15329494/107562846-a0d39800-6be0-11eb-9f61-227d52e757d2.png)
![onlydark](https://user-images.githubusercontent.com/15329494/107562850-a16c2e80-6be0-11eb-9beb-931afdc3883a.png)

#### Problem severity

High - broken application themes result in a very poor user experience!

#### Possible solutions

- introduce a real theming api in gnome
- use the gnome colors (not really a solution, as it collides with problem 3)

#### Problem severity

High - broken application themes result in a very poor user experience

#### Possible solutions

- introduce a real theming api in gnome
- remove the mixed default theme and only use a fully dark and a fully light applications theme (pictures in solution for 4b)

-------

I hope I could give a good top-down overview of the situation - please let me know what you think, all those problems and solutions echo in my brain since several years and some of you needed to hear it over and over but mostly in fragments in chats.

CC
@kenvandine @didrocks @cassiocassio @flexiondotorg @clobrano @madsrh @ubuntujaggers @Jupiter007-43 @Muqtxdir @3v1n0

**Comment by clobrano** [comment]
Hey @Feichtmeier, this is a huge sum up, thanks for the hard work :+1:

**Comment by clobrano** [comment]
If I read it right, you are exposing two main problems

1. Technical theming limits, which means lack of "downstream" theme customization capabilities.
2. Suboptimal user experience due in part to the missing of a complete mode switch feature and the color combination Yaru is using with dark headerbar whatever the theme is (light or dark).


A definitive solution to point 1 is "Create a theming API".
- There is already a door opened during Guadec 2019 by Yaru, ElementaryOS, and Gnome developers about this, ([here](https://wiki.gnome.org/GUADEC/2019/Hackingdays/VendorThemes/Notes) the outcomes). It is not what I would call an "API", more - as you said - an agreement, but it should work.

I tried to move to this field extending a Gnome proposal by Adrien Plazas (more [here](https://blog.carlolobrano.com/posts/2020-08-17-coloring-api-part1/)). I must admit I totally failed here, partially due to the free time available for sure, but mostly for the lack of knowledge (and working at weird times I did not want to bother Gnome devs during nights and out of their working times, my bad).

Problem 2 is part a theme problem and part a lack of feature.
- as lack of feature, we "just" need to bring the functionality to switch variants easily. I think ElementaryOS did it right, switching only the applications that "opt-in" to a dark variant. We should not force a dark mode in applications that do not propose it (and so, do not test it), as we do not force a light variant to applications that ship dark by default.
- as theme problem, it arised from the desire to adhere closer to upstream Gnome, which is good for many reasons (maintainance is one), but - from my point of view - at the beginning we flattened out on what Gnome offered, and we lost the design we developed to support a dark headerbar.
Right now the dark headerbar is still a distictive trait of Ubuntu theme, but brings some visibility issues. Disclaimer, I am not using it anymore.

I believe that Yaru should jump on the "Create a theming API" wagon. That's it. This would solve all the problems, but not in a near future.

**Reply by elsiehupp** [reply]
Thank you for the links to previous discussions!

My understanding regarding libadwaita's stance on [Adrien Plazas' proposal for a recoloring API](https://aplazas.pages.gitlab.gnome.org/blog/blog/2020/04/02/coloring-api.html) is that some of the maintainers [vehemently consider OEM-oriented system-wide recoloring to be beyond the scope](https://archive.is/oBFqZ).

The compromise way I might suggest of approaching the issue of accent color (which I describe in the above-linked thread) is allowing a system-wide accent color to be chosen from the [GNOME color palette](https://gitlab.gnome.org/World/design/palette). That is, Ubuntu could ship with the accent (i.e. "selection") color set to `GTK_STYLE_COLOR_ORANGE`, which libadwaita (or whatever the relevant library is) would provide, with widget shading, etc., all preconfigured. Providing an system-wide accent-color `gsettings` key-value pair (especially with an `enum` value rather than a hex value) would reduce the extent to which derivative themes would need to add or hack in additional functionality. (Because Yaru apparently uses several different accent colors, there may need to be multiple key-value pairs in `gsettings`, but the same basic principle of exposing it through `gsettings` still applies.)

An additional benefit of using an `enum` referring to a fixed palette is that the precise colors used could be defined to vary in saturation and lightness in the main four theme variations, `Adwaita`, `Adwaita-dark`, `HighContrast`, and `HighContrastInverse`, which would reduce the complexity required for developers to target all four official GTK themes.

The main limitation of using an `enum` referring to a fixed palette is that the precise hues available might not match a particular branded palette, but the flip side of this limitation is that using a fixed palette could improve the harmony between application-specified colors and system-specified colors.

On a somewhat more intensive level, hypothetically a theme (possibly even fully forked version of libadwaita) could redefine the color values provided by the GTK functions in question, so that the entire palette might be, e.g., flourescent, pastel, etc. If Yaru's implementation of Ubuntu/Canonical branding were decided to depend on using specific hues of orange, purple, etc., the values of these hues could be changed on a system-wide basis. Judging by the culture of GTK and libadwaita, changing the hues would probably require forking the relevant libraries, but the hue definitions could just as easily be exposed through `gsettings` if the maintainers felt inclined to do so.

Apologies if this is not the best place for me to reply with this comment. Finding the best place to talk about these things has been a bit of a wild goose chase...

**Comment by ubuntujaggers** [comment]
Thanks for the epic write-up @Feichtmeier :+1: 

My only counter-arguments are on these bits:

_"android and ios define what are minimum features for a modern desktop and people got very used to dark/light modes and expect it form a modern operating system!"_

Personally, I don't think the typical user will base their expectations of a desktop OS on Android or iOS.

I think there are still some fundamental differences between mobile devices and PCs, in terms of typical use.  Most people use their laptops and desktops indoors, during the day.  Either the lighting is optimised for productivity (at the office) or they control the lighting themselves (at home).

People use smartphones in a much bigger range of places (in bed, on the bus) and in places where they have no control over the light (a sunny park, a romantic restaurant).  So I suspect that a bigger chunk of users engage with light/dark themes on mobile phones compared to laptops.

I said before I think it's mainly power users who care about dark themes because they're doing computer stuff late at night. On reflection, I think the average person probably uses their smartphone like coders use their PCs. My parents don't code late at night, but I bet they read text messages in the dark!

Speaking personally, I'm always changing my phone's brightness setting, but when I'm on my laptop, I'm far more likely to turn the light on or close the curtains.

There are other differences on mobile. For instance: there's less need to divide the screen into distinct panels because fewer things are in view at once.  One of the most popular ways to organise a desktop UI is to have dark-on-light for the main area and then smaller areas that are inverted.  On desktop, the heading of an app is also a handle you can "grab" to move it around, so there's value in making it stand out from the body of the app. IMO, people's expectations of desktop UIs will be less influenced by Android and more by stuff like Office, Teams - even the Amazon web UI - where the mixed approach is still commonplace:

![image](https://user-images.githubusercontent.com/38893390/107676651-48a89e80-6c91-11eb-8cca-bb08c333fb2f.png)
![image](https://user-images.githubusercontent.com/38893390/107676744-6544d680-6c91-11eb-8a0e-59f4c8a8f359.png)
![image](https://user-images.githubusercontent.com/38893390/107677839-97a30380-6c92-11eb-8813-b36a6f6491e7.png)

I bet some newcomers _are_ surprised by the default Ubuntu theme, but I honestly think it's because of the jazzy colour scheme, rather than the mix of dark and light elements?

_"Medium - users not familiar to the history of Ubuntu themes are not known to the concept of "mixed" themes and would expect the theme to be either dark or light [...] remove the mixed themed and provide only dark and light application themes"_

Based on my previous comments, for me, the argument in favour of retiring the default (mixed) theme is purely the fact that it's apparently really hard to make an inverted title bar work, in our current Gnome world.

And tbh, that's already a good argument.  I love the current default theme and would prefer to keep it, but I also like the all-light variant, and it makes a lot of problems go away.  I can even imagine an all-light variant that I'd love, but we're constrained by having to stay close to upstream (basically, I don't like the way Adwaita - and therefore Yaru Light - distinguishes between foreground and background).

I'm less convinced by the argument that we should have all-light and all-dark only because that's what Android and iOS are doing.  For me, the acid test is to think how the user community would respond to the announcement.  If we said _"We're getting rid of inverted header bars because users will expect Ubuntu to be like Android and iOS,"_ then some people would be happy and some people would be sad, but I think a lot of people would just go _":facepalm: why are devs obsessed with making desktop UIs look like smartphone UIs nowadays."_

Whereas if we said _"we're getting rid of inverted header bars because it's a big departure from Adwaita, which has unexpected consequences for some apps and makes them look awful"_ then I think people would be far more understanding?

Also... I personally feel like problem 1 is actually "High".  All these problems flow from the fact that, when it comes to visual styling, important choices are made by people who aren't responsible for (or beholden to) Ubuntu.

**Reply by elsiehupp** [reply]
>> android and ios define what are minimum features for a modern desktop and people got very used to dark/light modes and expect it form a modern operating system!
>
> Personally, I don't think the typical user will base their expectations of a desktop OS on Android or iOS.

As someone who has spent significant amounts of time on both macOS and Windows, I can say that **the option for** "Dark Mode", however divisive "Dark Mode" itself may be, is increasingly the norm. So I would describe a high-level non-breaking implementation of system-wide Dark Mode less in terms of the merits of Dark Mode itself and more in terms of feature parity with major competitors.

Full disclosure, though: I immediately turned on Dark Mode when I started using Ubuntu, and I use "Night Light" despite [its lack of proven benefits](https://www.theverge.com/2021/5/6/22423239/apple-iphone-night-shift-blue-light-sleep) and despite the fact that it makes me screen look like it's covered in a thin film of kitchen grease. It's probably an [autistic sensory thing](https://www.verywellhealth.com/autism-and-sensory-overload-259892): my brother uses accessibility features to stain his screen a deep, dark orange 24/7.

**Reply by elsiehupp** [reply]
It may also be worth mentioning that GNOME at least seems to be trying to move in the direction of a fully adaptive interface (which is part of why they've implemented some controversially touch-oriented changes to the underlying UI paradigm). The Unity-based Ubuntu Touch may be severed from Canonical's imprimatur (and funding), but GNOME would really like you to load up GNOME on your [PinePhone](https://www.pine64.org/pinephone/) [\[1\]](https://fossbytes.com/linux-based-pinephone-pinetab-running-gnome-3-36/)[\[2\]](https://www.youtube.com/watch?v=w2owj4549_s), so Yaru on a smartphone isn't completely unimaginable, even if it isn't explicitly on Canonical's roadmap.

**Comment by Feichtmeier** [comment]
I can Screencast from a MacBook tomorrow. They follow the trend and offer a similar approach on light/dark mode. And I'm very sure Windows 10 will follow soon.

What you describe as coloured headerbars are still themes that have either the light or the dark attribute assigned to them. "This is a light theme". So when the system flips and the app opted in it can flip too.

I must admit that I personally don't like the idea of colouring a whole block in an app with a branding colour, but this is okay as long as it could be determined as a light or a dark theme.

What we do is something different. We overwrite all windows to have a dark headerbar. GTK apps though still assume that they are put into a light window.
So even if an app would have their branded headerbar, it would still be dark.
I think this is both wrong from a design perspective and technicaly.

**Comment by ubuntujaggers** [comment]
@Feichtmeier 

_"I can Screencast from a MacBook tomorrow. They follow the trend and offer a similar approach on light/dark mode. And I'm very sure Windows 10 will follow soon."_

I've actually got a Macbook in the flat - and it's true that Windows 10 might follow - but it doesn't _necessarily_ mean that we have to copy them.

If you're using lots of windows and some are in windowed mode, there are actually some benefits to a contrasting headerbar. E.g., compare these:

**Yaru light**

![image](https://user-images.githubusercontent.com/38893390/107707429-bff22880-6cb9-11eb-93ec-ccd88a64c69d.png)

**Yaru dark**

![image](https://user-images.githubusercontent.com/38893390/107707468-d4362580-6cb9-11eb-8de5-6115b039d350.png)

With this:

**Yaru default**

![image](https://user-images.githubusercontent.com/38893390/107707598-03e52d80-6cba-11eb-9f81-9f8e59fcc403.png)

If you've got a several apps open and some are in windowed mode, it's easier to spot the file manager app and bring it to the front or move it around when the window headers are visible. Obviously it's not a problem on mobile because each app would fill the screen and you'd be swiping between them.  It's also not a problem if your desktop workflow is analogous to that.

Personally, I often have one window maximised with others in windowed mode, and bring the little ones to the front and move them around when I need to compare things (especially when I'm doing graphics or making notes on something). Funnily enough, one reason I don't like MacOS very much is because it's not set up to make that easy :shrug: But that's another issue. IMO, it's easy for the desktop to turn into a wall of white (or a wall of black) if there aren't areas of contrast in the UI to help your eyes find things.

**But** I'm not saying I'm right or that the Mac approach is wrong. I just don't think we should take it for granted that people are going to come to Ubuntu expecting an all-light mode and and all-dark mode and be disappointed by the mixed mode, because that's what Android does and now Mac are doing the same, and they expect us to do it too.  Especially if there's actually an all-light and all-dark option as well.

Except...

_"What we do is something different. We overwrite all windows to have a dark headerbar. GTK apps though still assume that they are put into a light window.
So even if an app would have their branded headerbar, it would still be dark."_

Yep, I can't fault this point - the actual implementation of the dark header bars in Yaru causes issues that would go away if we didn't bother.  It's definitely hacky, right?

**Comment by Jupi007** [comment]
Thank you for this very complete presentation @Feichtmeier 👍

I agree in all points with @ubuntujaggers about the headerbar.
The inverted headerbar is just perfect in term of design and usability. It is pretty cute, it gives a real identity to the Ubuntu desktop and It offer an area with a high contrast that makes it easy to spot windows. And I don't think that many peoples think that our default theme is not a real light theme (in fact before joining you and discovering the joys of Adwaita, I thought the dark headerbar was the norm ^^).
But it also causes shame issues for an OS with a professional look (the best example is the baobab one - which is a preinstalled app moreover)... And the code is painful to manage as this is just hacking.
So my own opinion on this is ... that I haven't got an opinion 😑

About the shell, as the top bar is dark I think put the dark shell as default is a good idea. It's very classy IMO.

In fact the source of all problems stems from the first one: Ubuntu isn't use his own technologies for his desktop.

I never daily used another OS than Ubuntu since my 7 years old and it was ... Windows XP ^^ ... so I can't have really points of comparison with other modern systems.

**Comment by Feichtmeier** [comment]
Another example for why the mixed theme is a little bit problematic is that if we would someday provide an "auto" option depending on for example the environment light tracked by the webcam or just a time schedule, that our current three themes would not work. Either we would need to kill the light theme and say the mixed theme is the light theme by definition, or we would need to kill the mixed theme, or we would leave out the mixed theme as default and not selectable here and just as an option later

CC @elioqoshi  @waynecrosby

![grafik](https://user-images.githubusercontent.com/15329494/111764180-3b3f8f00-88a3-11eb-92ca-199561f69e68.png)
(MacOS installer)

**Reply by elioqoshi** [reply]
@Feichtmeier the 2nd option you mentioned seems more sensible to me. That's a problem for another day though as we can't have both fully light and dark themes yet, right?

**Reply by Feichtmeier** [reply]
> @Feichtmeier the 2nd option you mentioned seems more sensible to me. That's a problem for another day though as we can't have both fully light and dark themes yet, right?

Yes option two would be better, but it looks impossible to remove it from a branding perspective, as far as I understood most people in Canonical and Yaru

**Reply by Feichtmeier** [reply]
https://github.com/canonical/ubuntu-desktop-installer/issues/84

relevant, because a change in Ubuntu desktop would mean a change in the installer :)

**Reply by elsiehupp** [reply]
This may be less user-friendly, but perhaps the preferred themes for daytime and nighttime could be exposed as configurable settings, i.e. through `gsettings` or even in `gnome-control-center` itself. A contrarian could even specify a light theme at night and a dark theme during the day! Probably the important thing in terms of the settings panel would be avoiding excessive complexity, though obviously the issue of making system-wide color overrides actually work is much more immediate.

**Reply by elsiehupp** [reply]
I really need to learn how to make GTK interface mockups...

**Reply by elsiehupp** [reply]
Regarding the `gnome-control-center` Dark Mode UI, you can find the existing Gitlab Issues [here](https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/253) and [here](https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/1043), and you can find the further upstream `libadwaita` Gitlab Issue [here](https://gitlab.gnome.org/GNOME/libadwaita/-/issues/52).

**Comment by Muqtxdir** [comment]
Hi @ubuntu/yaru-team @elsiehupp @elioqoshi  👋

I am very late to answer for this discussion and I believe by now most of the annoying issues (*mixed-theme headerbar*) have been addressed by the removal of **standard-theme**. Now currently this repository only offers **Yaru** (*Light theme/Default theme*) and **Yaru-dark** (*Dark-theme*).

The only highly affected issue remaining is:
## 4a: ***the user preference for a dark or a light mode is not respected globally***
## Proposed solution:

Based on the discussions from discord and telegram
### 1) GNOME Shell:

- Add dark-theme patch to ubuntu-desktop's **_GNOME Shell_** to respect **-dark** preference (*already discussed still needs a yes from @3v1n0*).
- Modify the `gnome-shell-theme.gresource` according to the patch and revert back the default  GNOME Shell theme to **Yaru** instead of **Yaru-dark**
- **Screen-record** demo of the patched **GNOME Shell** and modified `gnome-shell-theme.gresource` below on ***Settings: Appearance***

https://user-images.githubusercontent.com/54065734/124395582-85e5f980-dd22-11eb-817a-6a8098166729.mp4

- **Screen-record** demo of the patched **GNOME Shell** and modified `gnome-shell-theme.gresource` below on ***GNOME Tweaks***


https://user-images.githubusercontent.com/54065734/124395616-b463d480-dd22-11eb-8a9e-0700f902fb02.mp4

- ***Note:***  If someone prefers having a dark **GNOME Shell** theme and light **GTK** theme or viceversa, they can do so by installing [user-themes-extension](https://extensions.gnome.org/extension/19/user-themes/).


https://user-images.githubusercontent.com/54065734/124395694-17ee0200-dd23-11eb-9e37-b9625f20f48d.mp4

### 2) Gedit color-scheme (issues: #2924, #2732)   

- Gedit currently is defaulted to use the **classic** `gtskourceview` and raises readibilty issues in **dark** theme.
- Install the **Yaru** and **Yaru-dark** `gtksourceview` files.
- Switch between light and dark themes based on **GTK** theme from ***Settings: Appearance*** by triggering `/org/gnome/gedit/preferences/editor/scheme`
- **Screen-record** demo of Gedit `gtksourceview` switching according to **GTK** theme

https://user-images.githubusercontent.com/54065734/124188293-b4729300-dadc-11eb-8071-f7c0c0c22400.mp4

Thanks! hope to see this being implemented in ubuntu-desktop 21.10 ^ ^

**Reply by Feichtmeier** [reply]
Thanks @Muqtxdir ! Looking forward to this in ubuntu desktop :) And very nice post!

**Reply by clobrano** [reply]
> 4a: _**the user preference for a dark or a light mode is not respected globally**_

Hi all, I've been absent lately, so I probably arrive late and you've already discussed this detail, if that so, that's fine.

Regarding syncing shell and gtk variants, that means having only 2 options
- light gtk and light shell
- dark gtk and dark shell

while having a light shell in a dark gtk has always been a reason for people to complain (too bright), I think there are many people used to have a dark shell and light gtk, and few of them would switch to "all dark" just to avoid the light shell.

Since mixed shell and gtk variants doesn't seem to be really appreciated here, I would at least suggest to add a switch in settings to "keep gtk and shell in sync", which - disabled - would allow to change gtk theme only

**Reply by 3v1n0** [reply]
Is this something relevant? I mean, we'd still want to keep the shell follow the system in dark/light preferences?

My impression was that we still wanted the shell widgets always dark (or always light) depending on what's the default is
